command-line-interface.md

```diff
@@ -0,0 +1,189 @@
+# Operations
+
+A conformant [runtime][] MUST provide an executable (called `funC` in the following examples).
```

Comment:
On Thu, Jul 28, 2016 at 12:40:07PM -0700, Doug Davis wrote:
> whether *we* define any other API or not isn't the point. People can
> develop any API/interface they want for their runtime - that's out
> of scope for us - but that doesn't mean they can't be compliant to
> the runtime. All they need to be able to do is adhere to the MUSTs
> in the runtime spec.
>
> When you have a set of specs people will often want to support a
> subset of them as it fits their needs. We don't want to force
> someone to either be non-compliant or to support something they
> don't want/need.

There are two levels to “compliant”.  From this spec [1]:

  An implementation is not compliant if it fails to satisfy one or
  more of the MUST or REQUIRED requirements for the protocols it
  implements. An implementation that satisfies all the MUST or
  REQUIRED and all the SHOULD requirements for its protocols is said
  to be "unconditionally compliant".

and the charter's “certification program” [2].  Folks who want to pick
and choose from MUST-level spec requirements are fine, but they're not
going to be compliant.  Folks who check off (say) all the config.md
and config-linux.md MUST-level requirements but skip this API may be
compliant with those parts (protocols?), but will not be OCI certified
if they don't expose an API that the compliance test understands.  At
the moment, this command line API is the only such API (and ocitools
certainly expects a command line API for runtime validation).

Runtimes that are compliant but not certified are going to be hard to
objectively identify.

[1]: https://github.com/opencontainers/runtime-spec/tree/v1.0.0-rc1#open-container-runtime-specification
[2]: https://www.opencontainers.org/about/governance